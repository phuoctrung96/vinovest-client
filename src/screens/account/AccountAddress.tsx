import React, { useState } from "react";
import styled from "styled-components";
import Skeleton from "react-loading-skeleton";
import { useTranslation } from "react-i18next";
import UserInformation from "./components/UserInformation";
import UserForm from "./components/UserForm";
import { CheckCircle2 } from "#assets/home";
import AlertMessage from "#shared/ui/Alert/AlertMessage";
import { PageHeader } from "#shared/components/PageHeader";

export interface AccountProps {
    fetchData: () => Promise<void>;
}
interface AccountAddressProps extends AccountProps {
    query: {
        firstName: string;
        lastName: string;
        country: string;
        address: string;
        city: string;
        state: string;
        zip: string;
    };
}

export const AccountAddress = ({ query, fetchData }: AccountAddressProps) => {
    const { t } = useTranslation("account");
    const [editState, setEditState] = useState(false);
    const toggleEditor = () => setEditState(!editState);
    const [submitComplete, updateSubmitComplete] = useState(false);

    return (
        <>
            {!query ? (
                <Skeleton style={{ height: "48px", width: "50%", marginBottom: "42px" }} />
            ) : (
                <PageHeader>{t("account_personal_information.title")}</PageHeader>
            )}
            {editState ? (
                <UserForm
                    onSubmitComplete={() => updateSubmitComplete(true)}
                    onClick={toggleEditor}
                    query={query}
                    fetchData={fetchData}
                />
            ) : (
                <>
                    {query && <EditLink onClick={toggleEditor}>{t("myaccount.edit")}</EditLink>}
                    <UserInformation query={query} />
                    {submitComplete && (
                        <AlertMessage type="success">
                            <CheckCircle2 />
                            {t("account_personal_information.success")}
                        </AlertMessage>
                    )}
                </>
            )}
        </>
    );
};

const EditLink = styled.span`
    color: #a86d37;
    cursor: pointer;
    font-size: 0.875rem;
    padding: 0.5rem 1rem;
    position: absolute;
    right: 0;
    text-transform: uppercase;
    top: 2rem;

    ${(p) => p.theme.media.greaterThan("1024px")`
        right: 2rem;
        top: 4.5rem;
    `};
`;
