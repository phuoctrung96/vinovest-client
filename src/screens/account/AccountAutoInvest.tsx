import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";
import { useTranslation } from "react-i18next";
import { PageHeader } from "#shared/components/PageHeader";

interface AccountAutoInvestProps {
    portalURIFetch: {
        data: string;
        status: {
            isSuccess: () => Promise<void>;
        };
    };
}

export const AccountAutoInvest = ({ portalURIFetch }: AccountAutoInvestProps) => {
    const { t } = useTranslation(["account"]);

    return (
        <>
            <PageHeader>{t("account_settings.auto_invest")}</PageHeader>

            <div>
                <div>{t("account_auto_invest.subtitle")}</div>

                <ButtonPrimary
                    className={!portalURIFetch.status.isSuccess() ? "disabled" : ""}
                    href={portalURIFetch.data}
                >
                    {t("account_auto_invest.title")}
                </ButtonPrimary>
            </div>
        </>
    );
};

AccountAutoInvest.propTypes = {
    portalURIFetch: PropTypes.shape({
        status: PropTypes.shape({
            isSuccess: PropTypes.func.isRequired,
        }),
        data: PropTypes.string,
    }).isRequired,
};

const ButtonPrimary = styled.a`
    font-family: VinovestMono;
    font-style: normal;
    font-weight: 500;
    line-height: 18px;
    letter-spacing: 0.025em;
    background: #a86d37;
    border: 1px solid #a86d37;
    color: #fff;
    cursor: pointer;
    display: inline-block;
    font-size: 0.875rem;
    margin-top: 1.5rem;
    min-width: 150px;
    padding: 1rem 2rem;
    text-align: center;
    text-decoration: none;
    text-transform: uppercase;
    width: 100%;

    ${(p) => p.theme.media.greaterThan("768px")`
        width: auto;
    `};

    &.disabled {
        background: #e5e5e5;
        border: 1px solid #e5e5e5;
        cursor: not-allowed;
    }
`;
