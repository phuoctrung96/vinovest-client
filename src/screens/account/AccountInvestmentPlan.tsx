import React, { useState, useEffect } from "react";
import styled from "styled-components";
import { withOktaAuth } from "@okta/okta-react";
import Skeleton from "react-loading-skeleton";
import { useTranslation } from "react-i18next";
import { OktaAuth } from "@okta/okta-auth-js";
import AlertMessage from "#shared/ui/Alert/AlertMessage";
import { CheckCircle2 } from "#assets/home";
import UserInvestmentPlan from "./components/UserInvestmentPlan";
import UserInvestmentPlanForm from "./components/UserInvestmentPlanForm";
import AccountService from "#services/UserService";
import { useRootStore } from "#shared/hooks";
import { PageHeader } from "#shared/components/PageHeader";

interface AccountInvestmentPlanProps {
    oktaAuth: OktaAuth;
}

const AccountInvestmentPlan = ({ oktaAuth }: AccountInvestmentPlanProps) => {
    const [editState, setEditState] = useState(false);
    const [submitComplete, updateSubmitComplete] = useState(false);
    const [query, setQuery] = useState(null);
    const toggleEditor = () => setEditState(!editState);
    const store = useRootStore();
    const { t } = useTranslation("account");
    const fetchData = async () => {
        await store.user.requestUserDetailsFromOkta(oktaAuth);
        const response = await AccountService.useAccountPortfolioGet(
            store.user.oktaUserEntity.data.sub,
            store.auth.accessToken,
        );
        store.user.fetchProfile();
        setQuery(response);
    };

    useEffect(() => {
        fetchData();
        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, []);
    return (
        <>
            {query === null ? (
                <Skeleton style={{ height: "48px", width: "50%", marginBottom: "42px" }} />
            ) : (
                <PageHeader>{t("account_investment_plan.title")}</PageHeader>
            )}
            {editState ? (
                <UserInvestmentPlanForm
                    onSubmitComplete={() => updateSubmitComplete(true)}
                    onClick={toggleEditor}
                    query={query}
                    fetchData={fetchData}
                />
            ) : (
                <>
                    {query && <EditLink onClick={toggleEditor}>{t("myaccount.edit")}</EditLink>}
                    <UserInvestmentPlan query={query} />
                    {submitComplete && (
                        <AlertMessage type="success">
                            <CheckCircle2 />
                            {t("account_investment_plan.updated")}
                        </AlertMessage>
                    )}
                </>
            )}
        </>
    );
};
export default withOktaAuth(AccountInvestmentPlan);

const EditLink = styled.span`
    color: #a86d37;
    cursor: pointer;
    font-size: 0.875rem;
    padding: 0.5rem 1rem;
    position: absolute;
    right: 0;
    text-transform: uppercase;
    top: 2rem;

    ${(p) => p.theme.media.greaterThan("1024px")`
        right: 2rem;
        top: 4.5rem;
    `};
`;
