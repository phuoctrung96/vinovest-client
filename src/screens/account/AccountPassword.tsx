import React, { useState } from "react";
import styled from "styled-components";
import Skeleton from "react-loading-skeleton";
import { useTranslation } from "react-i18next";
import AlertMessage from "#shared/ui/Alert/AlertMessage";
import { CheckCircle2 } from "#assets/home";
import { PageHeader } from "#shared/components/PageHeader";
import UserPasswordForm from "./components/UserPasswordForm";

export const AccountPassword = () => {
    const { t } = useTranslation("account");
    const [editState, setEditState] = useState(false);
    const [submitComplete, updateSubmitComplete] = useState(false);
    const toggleEditor = () => setEditState(!editState);

    return (
        <>
            <PageHeader>{t("password.title")}</PageHeader>

            {editState ? (
                <UserPasswordForm onSubmitComplete={() => updateSubmitComplete(true)} onClick={toggleEditor} />
            ) : (
                <>
                    <EditLink onClick={toggleEditor}>{t("myaccount.edit")}</EditLink>
                    <Box>
                        <span>{t("password.title")}</span>
                        <Populated type="password" value="notarealpassword" readOnly />
                    </Box>
                    {submitComplete && (
                        <AlertMessage type="success">
                            <CheckCircle2 />
                            {t("password.success_update")}
                        </AlertMessage>
                    )}
                </>
            )}
        </>
    );
};

const EditLink = styled.span`
    color: #a86d37;
    cursor: pointer;
    font-size: 0.875rem;
    padding: 0.5rem 1rem;
    position: absolute;
    right: 0;
    text-transform: uppercase;
    top: 2rem;

    ${(p) => p.theme.media.greaterThan("1024px")`
        right: 2rem;
        top: 4.5rem;
    `};
`;
const Box = styled.div`
    align-items: center;
    border-bottom: 1px solid #eeeeee;
    color: #766a7f;
    display: flex;
    font-size: 0.875rem;
    justify-content: space-between;
    margin-bottom: 1rem;

    span {
        padding-right: 2rem;
        white-space: nowrap;
    }

    div {
        font-size: 1rem;
        font-weight: 500;
        padding: 1rem 0;
        text-align: right;
        word-break: break-all;
    }
`;

const Populated = styled.input`
    border: 0;
    color: ${(p) => p.theme.colors.mainAccentBlue};
    flex: 1;
    padding: 1rem 0;
    pointer-events: none;
    text-align: right;
`;
