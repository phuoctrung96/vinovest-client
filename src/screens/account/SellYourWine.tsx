import React from "react";
import styled from "styled-components";
import { observer } from "mobx-react-lite";
import Skeleton from "react-loading-skeleton";
import { useTranslation } from "react-i18next";
import posthog from "posthog-js";
import { useHistory } from "#shared/hooks/useHistory";
import { MainButton } from "#shared/ui";
import { ROUTE_PATHS } from "#screens/route-paths";
import { getStatus } from "#models/FetchStatus";
import { useRootStore } from "#shared/hooks";
import { PageHeader } from "#shared/components/PageHeader";
import {I18nLink as Link} from "#localization/localizedRouter";
import { useConfig } from "#shared/hooks";
import { languageCodeChina } from "../../utils/constants";

export const SellYourWine = observer(() => {
    const { t } = useTranslation(["account"]);
    const config = useConfig();

    const s = useRootStore();
    const currencyCode = s.user.userCurrency;
    const history = useHistory();
    const winesInLiquidationProcessIsDone = getStatus(s.liquidation.winesInLiquidationProcessEntity).isDone();
    const liquidateWinesListIsDone = getStatus(s.liquidation.liquidateWinesListEntity).isDone();

    const { winesInLiquidationProcess, liquidateWinesList } = s.liquidation;

    React.useEffect(() => {
        if (s.auth.isAuthenticated && s.user.oktaUserInfo && s.user.oktaUserInfo.sub) {
            if (!s.liquidation.winesInLiquidationProcess.length) {
                s.liquidation.fetchWinesInLiquidationProcess();
            }

            if (!s.liquidation.liquidateWinesList.length) {
                s.liquidation.fetchLiquidateWinesList(currencyCode);
            }
        }

        return () => {
            s.liquidation.resetState();
        };
        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [s.auth.isAuthenticated, s.user.oktaUserInfo.sub]);

    const disableButton = winesInLiquidationProcess.length && liquidateWinesList.length;

    function handleSellYourWine() {
        try {
            posthog.capture("click_sell_your_wine_account_setting", {});
        } catch (err) {}
        history.push(ROUTE_PATHS.beforeYouRequest);
    }
    const onBookACall = React.useCallback(() => {
        posthog.capture("rebalance_calendly", {
            component: "SellYourWine",
            progress: "launch",
            location: "overview",
        });
        if (!languageCodeChina) {
            window.Calendly.initPopupWidget({
                url: config.calendly.rebalancing
            })
        } else {
            window.location.href = "https://airtable.com/shr0yMiUh5ty4zjyP";
        }
    }, [config]);

    return (
        <>
            <Wrapper style={{ marginBottom: "20px" }}>
                {!winesInLiquidationProcessIsDone || !liquidateWinesListIsDone ? (
                    <Skeleton style={{ height: "48px", width: "50%", marginBottom: "42px" }} />
                ) : (
                    <PageHeader>{t("account_settings.rebalance")}</PageHeader>
                )}
                {!winesInLiquidationProcessIsDone || !liquidateWinesListIsDone ? (
                    <>
                        <Skeleton count={1} height="50px" style={{ marginBottom: "1.5rem" }} />
                        <Skeleton count={1} style={{ height: "50px", width: "300px", marginBottom: "1.5rem" }} />
                    </>
                ) : (
                    <>
                        <Description style={{ marginBottom: "40px" }}>{t("rebalance_description")}</Description>
                        <BookACall
                            onClick={onBookACall}
                            disabled={disableButton || !s.liquidation.liquidateWinesList.length}
                        >
                            {t("book_a_call")}
                        </BookACall>
                    </>
                )}
            </Wrapper>

            <Wrapper>
                {!winesInLiquidationProcessIsDone || !liquidateWinesListIsDone ? (
                    <Skeleton style={{ height: "48px", width: "50%", marginBottom: "42px" }} />
                ) : (
                    <PageHeader>{t("account_settings.liquidate_portfolio")}</PageHeader>
                )}

                <div>
                    <Container>
                        {!winesInLiquidationProcessIsDone || !liquidateWinesListIsDone ? (
                            <>
                                <Skeleton count={1} height="50px" style={{ marginBottom: "1.5rem" }} />
                                <Skeleton
                                    count={1}
                                    style={{ height: "50px", width: "300px", marginBottom: "1.5rem" }}
                                />
                                <Skeleton count={1} height="28px" />
                            </>
                        ) : (
                            <>
                                <Description>{t("sell_your_wine.description.paragraph1")}</Description>
                                <Description>
                                    <p style={{ margin: "0" }}>{t("sell_your_wine.description.paragraph2")}</p>
                                    <p style={{ margin: "0" }}>{t("sell_your_wine.description.paragraph3")}</p>
                                </Description>
                                <Description>{t("sell_your_wine.description.paragraph4")}</Description>

                                <ButtonContainer>
                                    <MainSellButton
                                        onClick={handleSellYourWine}
                                        disabled={disableButton || !s.liquidation.liquidateWinesList.length}
                                    >
                                        {disableButton
                                            ? t("sell_your_wine.sell_button.enabled")
                                            : t("sell_your_wine.sell_button.disabled")}
                                    </MainSellButton>
                                    <ContactSupport>
                                        {disableButton ? (
                                            <>
                                                {t("sell_your_wine.support.sold")}{" "}
                                                <a href="mailto:support@vinovest.co">
                                                    {t("sell_your_wine.support.sold_email_message")}
                                                </a>
                                            </>
                                        ) : (
                                            <>
                                                {t("sell_your_wine.support.notsold")}{" "}
                                                <Link to="/help">
                                                    {t("sell_your_wine.support.notsold_email_message")}
                                                </Link>
                                            </>
                                        )}
                                    </ContactSupport>
                                </ButtonContainer>
                            </>
                        )}
                    </Container>
                </div>
            </Wrapper>
        </>
    );
});

const Container = styled.div`
    width: 100%;
`;

const Description = styled.div`
    font-size: 16px;
    line-height: 25px;
    color: #242424;
    padding: 5px;
    margin-top: 24px;

    @media screen and (max-width: 768px) {
        font-size: 14px;
        line-height: 21px;
    }
`;

const ButtonContainer = styled.div`
    margin-top: 0.5rem;

    button {
        margin: 8px auto;
    }
`;

const MainSellButton = styled(MainButton)`
    margin-left: 0 !important;
`;

const ContactSupport = styled.div`
    margin-top: 35px;
    width: 100%;
    text-align: left;
    font-family: VinovestMedium;
    font-style: normal;
    font-weight: 500;
    font-size: 16px;
    line-height: 26px;
    letter-spacing: 0.005em;
    color: #242e35;

    @media screen and (max-width: 768px) {
        font-size: 14px;
        line-height: 21px;
    }

    a {
        color: #a86d37;
        text-decoration: none;
    }
`;

const Wrapper = styled.div`
    border: 1px solid ${(p) => p.theme.colors.lighterGray};
    border-radius: 10px;
    box-shadow: 0px 4px 40px rgba(0, 0, 0, 0.1);
    padding: 1.5rem 1rem;
    ${(p) => p.theme.media.greaterThan("1024px")` 
        padding: 3rem;       
  `};
`;

const BookACall = styled(MainSellButton)`
    width: 220px;
    @media (max-width: 767px) {
        width: 100%;
    }
`;
