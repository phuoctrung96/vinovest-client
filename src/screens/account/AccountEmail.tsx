import React, { useState } from "react";
import styled from "styled-components";
import Skeleton from "react-loading-skeleton";
import { useTranslation } from "react-i18next";
import UserEmailForm from "./components/UserEmailForm";
import UserEmail from "./components/UserEmail";
import AlertMessage from "#shared/ui/Alert/AlertMessage";
import { CheckCircle2 } from "#assets/home";
import { PageHeader } from "#shared/components/PageHeader";
import { AccountProps } from "./AccountAddress";

interface AccountEmailProps extends AccountProps {
    query: {
        email: string;
    };
}

export const AccountEmail = ({ query, fetchData }: AccountEmailProps) => {
    const { t } = useTranslation("account");
    const [editState, setEditState] = useState(false);
    const [submitComplete, updateSubmitComplete] = useState(false);
    const toggleEditor = () => setEditState(!editState);

    return (
        <>
            {!query ? (
                <Skeleton style={{ height: "48px", width: "50%", marginBottom: "42px" }} />
            ) : (
                <PageHeader>{t("account_email.email")}</PageHeader>
            )}
            {editState ? (
                <UserEmailForm
                    onSubmitComplete={() => updateSubmitComplete(true)}
                    onClick={toggleEditor}
                    query={query}
                    fetchData={fetchData}
                />
            ) : (
                <>
                    {query && <EditLink onClick={toggleEditor}>{t("myaccount.edit")}</EditLink>}
                    <UserEmail query={query} />
                    {submitComplete && (
                        <AlertMessage type="success">
                            <CheckCircle2 />
                            {t("account_email.updated")}
                        </AlertMessage>
                    )}
                </>
            )}
        </>
    );
};

const EditLink = styled.span`
    color: #a86d37;
    cursor: pointer;
    font-size: 0.875rem;
    padding: 0.5rem 1rem;
    position: absolute;
    right: 0;
    text-transform: uppercase;
    top: 2rem;

    ${(p) => p.theme.media.greaterThan("1024px")`
        right: 2rem;
        top: 4.5rem;
    `};
`;
