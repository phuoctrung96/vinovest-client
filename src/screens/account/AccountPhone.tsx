import React, { useState } from "react";
import styled from "styled-components";
import { formatPhoneNumber } from "react-phone-number-input";
import Skeleton from "react-loading-skeleton";
import { useTranslation } from "react-i18next";
import { CheckCircle2 } from "#assets/home";
import { UserDisclaimer } from "./components/UserDisclaimer";
import AlertMessage from "#shared/ui/Alert/AlertMessage";
import UserPhoneForm from "./components/UserPhoneForm";
import { PageHeader } from "#shared/components/PageHeader";
import { AccountProps } from "./AccountAddress";

interface AccountPhoneProps extends AccountProps {
    query: {
        primaryPhone: string;
    };
}

export const AccountPhone = ({ query, fetchData }: AccountPhoneProps) => {
    const [submitComplete, updateSubmitComplete] = useState(false);
    const [editState, setEditState] = useState(false);
    const noResults = "Not provided";
    const toggleEditor = () => setEditState(!editState);
    const { t } = useTranslation("account");

    if (!query) {
        return (
            <>
                <Skeleton style={{ height: "48px", width: "50%", marginBottom: "42px" }} />
                <Skeleton count={1} height="51px" style={{ marginBottom: "1.5rem" }} />
                <Skeleton count={1} height="12px" />
            </>
        );
    }
    const formattedNumber = formatPhoneNumber(query.primaryPhone);

    return (
        <>
            <PageHeader>{t("account_mobile.title")}</PageHeader>

            {editState ? (
                <>
                    <UserPhoneForm
                        onSubmitComplete={() => updateSubmitComplete(true)}
                        onClick={toggleEditor}
                        query={query.primaryPhone}
                        fetchData={fetchData}
                    />
                </>
            ) : (
                <>
                    <EditLink onClick={toggleEditor}>{t("myaccount.edit")}</EditLink>
                    <Box>
                        <span>{t("account_mobile.label")}</span>
                        {formattedNumber ? (
                            <Populated>{formattedNumber}</Populated>
                        ) : (
                            <Unpopulated>{noResults}</Unpopulated>
                        )}
                    </Box>
                    <UserDisclaimer />
                    {submitComplete && (
                        <AlertMessage type="success">
                            <CheckCircle2 />
                            {t("account_mobile.success_updated")}
                        </AlertMessage>
                    )}
                </>
            )}
        </>
    );
};

const EditLink = styled.span`
    color: #a86d37;
    cursor: pointer;
    font-size: 0.875rem;
    padding: 0.5rem 1rem;
    position: absolute;
    right: 0;
    text-transform: uppercase;
    top: 2rem;

    ${(p) => p.theme.media.greaterThan("1024px")`
        right: 2rem;
        top: 4.5rem;
    `};
`;
const Box = styled.div`
    align-items: center;
    border-bottom: 1px solid #eeeeee;
    color: #766a7f;
    display: flex;
    font-size: 0.875rem;
    justify-content: space-between;
    margin-bottom: 1rem;

    span {
        padding-right: 2rem;
        white-space: nowrap;
    }

    div {
        font-size: 1rem;
        font-weight: 500;
        padding: 1rem 0;
        text-align: right;
        word-break: break-all;
    }
`;
const Populated = styled.div`
    color: #242e35;
    word-break: break-word;
`;

const Unpopulated = styled.div`
    color: #a8abad;
`;
