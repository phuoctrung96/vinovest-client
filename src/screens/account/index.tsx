import React, { useEffect, useState, useCallback } from "react";
import ReactDOM from "react-dom";
import { observer } from "mobx-react-lite";
import styled from "styled-components";
import { withOktaAuth } from "@okta/okta-react";
import { Route, useLocation } from "react-router-dom";
import { useTranslation } from "react-i18next";
import { OktaAuth } from "@okta/okta-auth-js";
import { PageRedirect } from "#shared/components/PageRedirect/PageRedirect";
import { useLogoutCallback, useRootStore } from "#shared/hooks";
import { AccountPhone } from "./AccountPhone";
import { AccountAddress } from "./AccountAddress";
import { AccountEmail } from "./AccountEmail";
import { SellYourWine } from "./SellYourWine";
import { AccountAutoInvest } from "./AccountAutoInvest";
import AccountInvestmentPlan from "./AccountInvestmentPlan";
import { I18nSwitch, I18nNavLink } from "#localization/localizedRouter";
import { AccountPassword } from "./AccountPassword";

interface AccountPageProps {
    oktaAuth: OktaAuth;
}

const AccountPage = ({ oktaAuth }: AccountPageProps) => {
    const { t } = useTranslation("account");
    const onLogout = useLogoutCallback();
    const store = useRootStore();
    const [userInformation, setUserInformation] = useState(null);
    const { transfer: transferStore, auth: authStore } = useRootStore();
    const location = useLocation();

    useEffect(() => {
        if (authStore.isAuthenticated) {
            transferStore.requestGetSelfServicePortalURI();
        }
    }, [authStore.isAuthenticated, transferStore]);

    const fetchData = useCallback(async () => {
        await store.user.requestUserDetailsFromOkta(oktaAuth);
        await store.user.fetchUserInformation();
        setUserInformation(store.user.userInformationEntity.data);
    }, [oktaAuth, store.user]);

    useEffect(() => {
        fetchData();
    }, [fetchData]);
    return (
        <AccountPageContainer>
            <NavigationPanel>
                <MenuTitle>{t("main.user_settings")}</MenuTitle>
                <I18nNavLink to="/account" activeClassName="active" exact>
                    {t("user_settings.account")}
                </I18nNavLink>
                <I18nNavLink to="/account/email" activeClassName="active">
                    {t("account_email.email")}
                </I18nNavLink>
                <I18nNavLink to="/account/password" activeClassName="active">
                    {t("user_settings.password")}
                </I18nNavLink>
                <I18nNavLink to="/account/phone" activeClassName="active">
                    {t("user_settings.phone")}
                </I18nNavLink>
                <MenuTitle>{t("main.account")}</MenuTitle>
                <I18nNavLink to="/account/investment-plan" activeClassName="active">
                    {t("user_settings.investment_plan")}
                </I18nNavLink>
                <I18nNavLink to="/account/auto-invest" activeClassName="active">
                    {t("account_settings.auto_invest")}
                </I18nNavLink>
                <I18nNavLink to="/account/sell" activeClassName="active">
                    {t("account_settings.sell_wine")}
                </I18nNavLink>
                <NavigationOption className="accent" onClick={onLogout}>
                    {t("user_settings.logout")}
                </NavigationOption>
            </NavigationPanel>
            <RightDetailPanel>
                <AccountFormWrapper loc={location.pathname}>
                    <I18nSwitch>
                        <Route
                            exact
                            path="/account"
                            render={(props) => (
                                <AccountAddress {...props} query={userInformation} fetchData={fetchData} />
                            )}
                        />
                        <Route
                            exact
                            path="/account/email"
                            render={(props) => (
                                <AccountEmail {...props} query={userInformation} fetchData={fetchData} />
                            )}
                        />
                        <Route exact path="/account/password" render={(props) => <AccountPassword />} />
                        <Route
                            exact
                            path="/account/phone"
                            render={(props) => (
                                <AccountPhone {...props} query={userInformation} fetchData={fetchData} />
                            )}
                        />
                        <Route exact path="/account/investment-plan" component={AccountInvestmentPlan} />
                        <Route
                            exact
                            path="/account/auto-invest"
                            render={(props) => (
                                <AccountAutoInvest {...props} portalURIFetch={transferStore.portalURIFetch} />
                            )}
                        />
                        <Route exact path="/account/sell" component={SellYourWine} />
                    </I18nSwitch>
                </AccountFormWrapper>
            </RightDetailPanel>
        </AccountPageContainer>
    );
};

export default withOktaAuth(observer(AccountPage));

const AccountFormWrapper = styled.div`
    border: ${(p) => (p.loc.includes("/account/sell") ? "none" : `1px solid ${p.theme.colors.lighterGray}`)};
    border-radius: 10px;
    box-shadow: ${(p) => (p.loc.includes("/account/sell") ? "none" : "0px 4px 40px rgba(0, 0, 0, 0.1)")};
    padding: 1.5rem 1rem;
    position: relative;

    ${(p) => p.theme.media.greaterThan("1024px")`
        margin: auto;
        max-width: 775px;
        padding: ${(p) => (p.loc.includes("/account/sell") ? "0" : "3rem")};
  `};
`;

const AccountPageContainer = styled.div`
    display: flex;
    flex-direction: column;
    padding: 20px;
    width: 100%;

    ${(p) => p.theme.media.greaterThan("1024px")`
        flex-direction: row;
        margin: auto;
        max-width: 1240px;
        padding: 80px 12px;
  `};
    ${(p) => p.theme.media.greaterThan("1100px")`
        flex-direction: row;
  `};
`;

const NavigationPanel = styled.div`
    display: none;
    overflow: auto;
    font-family: VinovestMono;
    font-size: 14px;
    justify-content: flex-start;
    margin: 24px 0;
    padding-bottom: 18px;
    border-bottom: 1px solid ${(p) => p.theme.colors.lightGray};

    ${(p) => p.theme.media.greaterThan("1024px")`
    display: flex;
        flex-direction: column;
        justify-content: flex-start;
        margin: initial;
        border-bottom: none;
        height: 100%;
        width: 300px;
        padding: 20px 0;
    `};

    a {
        background: none;
        border-left: 2px solid transparent;
        color: ${(p) => p.theme.colors.mainAccentBlue};
        white-space: nowrap;
        outline: none;
        border: none;
        cursor: pointer;
        text-decoration: none;
        text-transform: uppercase;
        margin-bottom: 1.5rem;
        padding: 0.25rem 0.75rem;
        text-align: left;

        &:hover {
            border-left: 2px solid ${(p) => p.theme.colors.burntOrange};
        }

        &.active {
            border-left: 2px solid ${(p) => p.theme.colors.burntOrange};
        }
    }
`;

const NavigationOption = styled.button`
    background: none;
    border-left: 2px solid ${(p) => (p.emphasis ? p.theme.colors.burntOrange : "transparent")}
    color: ${(p) => p.theme.colors.mainAccentBlue};
    border: none;
    cursor: pointer;
    text-transform: uppercase;
    outline: none;
    margin-bottom: 1.5rem;
    padding: .25rem .75rem;
    text-align: left;
    white-space: nowrap;

    &:hover {
        border-left: 2px solid ${(p) => p.theme.colors.burntOrange}
    }

    &.accent {
        color: ${(p) => p.theme.colors.burntOrange};
    }
`;

const RightDetailPanel = styled.div`
    padding: 20px 0;
    width: 100%;

    ${(p) => p.theme.media.greaterThan("1024px")`
        padding: 20px 0 20px 40px;
    `};
`;

const MenuTitle = styled.span`
    display: none;
    font-size: 1.25rem;
    font-family: RoslindaleDisplayCondensed;
    font-weight: 500;

    ${(p) => p.theme.media.greaterThan("1024px")`
        display: block;
        margin-bottom: 1.5rem;
    `};
`;
