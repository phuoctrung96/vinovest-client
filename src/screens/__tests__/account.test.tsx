import React, { ReactDOM } from "react";
import { render, screen } from "@testing-library/react";
import { AccountAutoInvest } from "../account/AccountAutoInvest";
import Account from "../account/index";
import { withContexts } from "#shared/testing/wrappers";
import { account_auto_invest } from "#translations/en/account.json";
import { i18nFailRegex } from "../../shared/testing/i18n";

const { title: sTitle, subtitle: sSubtitle } = account_auto_invest;

jest.mock("@okta/okta-react", () => ({
    useOktaAuth: () => ({
        authState: { isAuthenticated: true },
        authService: { handleAuthentication: jest.fn() },
    }),
    withOktaAuth: (x) => x,
}));
xdescribe("AccountAutoInvest component", () => {
    let component;
    const props = {
        portalURIFetch: {
            status: {
                isSuccess: jest.fn(() => true),
                isInvalidated: jest.fn(() => true),
            },
            data: "hello",
        },
    };
    component = withContexts(AccountAutoInvest);

    it("Should match snapshot", () => {
        const { baseElement } = render(component(props));
        expect(baseElement).toMatchSnapshot();
    });
    it("Check Translations", () => {
        const { getByTestId } = render(component(props));
        const title = getByTestId("acc-auto-link");
        const subtitle = getByTestId("acc-subtitle");

        expect(title.textContent).toEqual(sTitle);
        expect(subtitle.textContent).toEqual(sSubtitle);
    });
});
describe("Full Component", () => {
    // FULL COMPONENT HERE
    const props = {
        editState: false,
        query: true,
        fetchData: jest.fn(),
    };

    const { component } = withContexts(Account);

    const { baseElement } = render('<div id="root"/>', component(props));
    it("Should Translate correctly", () => {
        expect(baseElement).toMatchSnapshot();

        const failedTranslations = screen.queryByText(i18nFailRegex);
        expect(failedTranslations).toBeNull();
    });
});
