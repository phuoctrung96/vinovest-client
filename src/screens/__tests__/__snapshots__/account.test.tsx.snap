// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Full Component Should Translate correctly 1`] = `
<body>
  <div>
    &lt;div id="root"/&gt;
  </div>
</body>
`;
